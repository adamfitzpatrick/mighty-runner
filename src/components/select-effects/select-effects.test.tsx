import { h } from 'preact'
import deep from 'preact-render-spy'
import SelectEffects from '.'
import { Effect } from '@models';

describe('SelectEffects component', () => {
  let parentStat: HTMLElement
  let effects: Effect[]
  let toggleHandler: (effectId: string) => void

  beforeEach(() => {
    effects = [{
      name: 'Effect 1',
      id: '1',
      value: 1,
      active: true,
      target: [ 'target', '1' ]
    }, {
      name: 'Effect 2',
      id: '2',
      value: 3,
      active: false,
      target: [ 'target', '1' ]
    }]
    toggleHandler = jest.fn()
  })

  test('should render correctly with no optional props', () => {
    const selectEffects = deep(
      <SelectEffects effects={effects} toggleHandler={toggleHandler} parentStat={parentStat} />,
      { depth: 2 }
    )
    expect(selectEffects).toMatchSnapshot()
  })

  test('should render correctly when visible', () => {
    const selectEffects = deep(
      <SelectEffects effects={effects} toggleHandler={toggleHandler} visible parentStat={parentStat} />,
      { depth: 2 }
    )
    expect(selectEffects).toMatchSnapshot()
  })

  test('should render correctly with the top prop', () => {
    const selectEffects = deep(
      <SelectEffects effects={effects} toggleHandler={toggleHandler} visible top parentStat={parentStat} />,
      { depth: 2 }
    )
    expect(selectEffects).toMatchSnapshot()
  })

  test('should render correctly with the left prop', () => {
    const selectEffects = deep(
      <SelectEffects effects={effects} toggleHandler={toggleHandler} visible left parentStat={parentStat} />,
      { depth: 2 }
    )
    expect(selectEffects).toMatchSnapshot()
  })

  test('should render correctly with the left & top props', () => {
    const selectEffects = deep(
      <SelectEffects effects={effects} toggleHandler={toggleHandler} visible top left parentStat={parentStat} />,
      { depth: 2 }
    )
    expect(selectEffects).toMatchSnapshot()
  })

  test('should render with sufficient height to contain all effects when required', () => {
    effects.concat([{
      name: 'Effect 3',
      id: '3',
      value: 3,
      active: false,
      target: [ 'target', '3' ]
    }, {
      name: 'Effect 4',
      id: '4',
      value: 4,
      active: true,
      target: [ 'target', '4' ]
    }])
    const selectEffects = deep(
      <SelectEffects effects={effects} toggleHandler={toggleHandler} visible parentStat={parentStat} />,
      { depth: 2 }
    )
    expect(selectEffects).toMatchSnapshot()
  })

  describe('when an effect is clicked', () => {
    test('should call the provided toggle handler with the effect ID string', () => {
      const selectEffects = deep(
        <SelectEffects effects={effects} toggleHandler={toggleHandler} visible parentStat={parentStat} />,
        { depth: 2 }
      )
      selectEffects.find('input').at(0).simulate('click')
      expect(toggleHandler).toHaveBeenCalledWith('1')
    })
  })
})
