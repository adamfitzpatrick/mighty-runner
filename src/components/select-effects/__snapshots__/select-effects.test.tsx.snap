// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectEffects component should render correctly when visible 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="effects visible"
  style="width: calc(undefinedpx + 14rem);"
>
  <div class="list">
    <label
      class="effectToggle"
      aria-label="Effect 1, value 1, active"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="1"
        onClick={[Function onClick]}
        checked={true}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator active"></span>
      <span class="effectName">Effect 1</span>
      <span class="effectValue">1</span>
    </label>
    <label
      class="effectToggle"
      aria-label="Effect 2, value 3, inactive"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="2"
        onClick={[Function onClick]}
        checked={false}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator"></span>
      <span class="effectName">Effect 2</span>
      <span class="effectValue">3</span>
    </label>
  </div>
  <div
    class="editButton"
    style="margin-left: calc(undefinedpx + 2rem);"
  >
    edit stat...
  </div>
</div>

`;

exports[`SelectEffects component should render correctly with no optional props 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="effects"
  style="width: 0px;"
>
  <div class="list">
    <label
      class="effectToggle"
      aria-label="Effect 1, value 1, active"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="1"
        onClick={[Function onClick]}
        checked={true}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator active"></span>
      <span class="effectName">Effect 1</span>
      <span class="effectValue">1</span>
    </label>
    <label
      class="effectToggle"
      aria-label="Effect 2, value 3, inactive"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="2"
        onClick={[Function onClick]}
        checked={false}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator"></span>
      <span class="effectName">Effect 2</span>
      <span class="effectValue">3</span>
    </label>
  </div>
  <div
    class="editButton"
    style="margin-left: calc(undefinedpx + 2rem);"
  >
    edit stat...
  </div>
</div>

`;

exports[`SelectEffects component should render correctly with the left & top props 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="effects visible left top"
  style="width: calc(undefinedpx + 14rem);"
>
  <div class="list">
    <label
      class="effectToggle"
      aria-label="Effect 1, value 1, active"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="1"
        onClick={[Function onClick]}
        checked={true}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator active"></span>
      <span class="effectName">Effect 1</span>
      <span class="effectValue">1</span>
    </label>
    <label
      class="effectToggle"
      aria-label="Effect 2, value 3, inactive"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="2"
        onClick={[Function onClick]}
        checked={false}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator"></span>
      <span class="effectName">Effect 2</span>
      <span class="effectValue">3</span>
    </label>
  </div>
  <div
    class="editButton"
    style="margin-right: calc(undefinedpx + 2rem);"
  >
    edit stat...
  </div>
</div>

`;

exports[`SelectEffects component should render correctly with the left prop 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="effects visible left"
  style="width: calc(undefinedpx + 14rem);"
>
  <div class="list">
    <label
      class="effectToggle"
      aria-label="Effect 1, value 1, active"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="1"
        onClick={[Function onClick]}
        checked={true}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator active"></span>
      <span class="effectName">Effect 1</span>
      <span class="effectValue">1</span>
    </label>
    <label
      class="effectToggle"
      aria-label="Effect 2, value 3, inactive"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="2"
        onClick={[Function onClick]}
        checked={false}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator"></span>
      <span class="effectName">Effect 2</span>
      <span class="effectValue">3</span>
    </label>
  </div>
  <div
    class="editButton"
    style="margin-right: calc(undefinedpx + 2rem);"
  >
    edit stat...
  </div>
</div>

`;

exports[`SelectEffects component should render correctly with the top prop 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="effects visible top"
  style="width: calc(undefinedpx + 14rem);"
>
  <div class="list">
    <label
      class="effectToggle"
      aria-label="Effect 1, value 1, active"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="1"
        onClick={[Function onClick]}
        checked={true}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator active"></span>
      <span class="effectName">Effect 1</span>
      <span class="effectValue">1</span>
    </label>
    <label
      class="effectToggle"
      aria-label="Effect 2, value 3, inactive"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="2"
        onClick={[Function onClick]}
        checked={false}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator"></span>
      <span class="effectName">Effect 2</span>
      <span class="effectValue">3</span>
    </label>
  </div>
  <div
    class="editButton"
    style="margin-left: calc(undefinedpx + 2rem);"
  >
    edit stat...
  </div>
</div>

`;

exports[`SelectEffects component should render with sufficient height to contain all effects when required 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="effects visible"
  style="width: calc(undefinedpx + 14rem);"
>
  <div class="list">
    <label
      class="effectToggle"
      aria-label="Effect 1, value 1, active"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="1"
        onClick={[Function onClick]}
        checked={true}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator active"></span>
      <span class="effectName">Effect 1</span>
      <span class="effectValue">1</span>
    </label>
    <label
      class="effectToggle"
      aria-label="Effect 2, value 3, inactive"
      aria-live={true}
    >
      <input
        type="checkbox"
        value="2"
        onClick={[Function onClick]}
        checked={false}
        class="effectInput"
       />
      <span class="effectFocusIndicator"></span>
      <span class="effectIndicator"></span>
      <span class="effectName">Effect 2</span>
      <span class="effectValue">3</span>
    </label>
  </div>
  <div
    class="editButton"
    style="margin-left: calc(undefinedpx + 2rem);"
  >
    edit stat...
  </div>
</div>

`;
